code/scripts/DataTableGen.hs

```diff
@@ -0,0 +1,292 @@
+-- FIXME: use real parser (Low Priority; see line 119)
+module DataTableGen (main) where
+
+import Data.List
+import Data.Maybe
+import System.IO
+import System.Directory
+import Control.Monad
+import qualified Data.Map as Map
+import qualified DirectoryController as DC (createFolder, createFile, finder, 
+  getDirectories, stripPath, DrasilPack, FileName, FolderName, File(..), 
+  Folder(..))
+import SourceCodeReader as SCR (extractEntryData)
+
+type FileInstance = String
+type IsInstanceOf = String
+
+type DataType = String
+type Newtype = String
+type DNType = String
+type ClassName = String
+type EntryString = String
+
+-- Entry data type for storing entry data for each file entry
+data Entry = Entry { drasilPack :: DC.DrasilPack
+                   , fileName :: DC.FileName -- change to entryName?
+                   , filePath :: FilePath -- change to entryPath?
+                   , dataTypes :: [DataType]
+                   , newtypes :: [Newtype]
+                   , classes :: [Class]
+                   , classInstances :: [ClassInstance]
+                   } deriving (Show)
+-- ClassInstance data type for storing data/newtype type + class instance names
+data ClassInstance = ClassInstance {dnType :: DNType, clsInstName :: ClassName} 
+  deriving (Show)
+-- Class data type to store class name + type
+data Class = Class {className :: ClassName, classType :: ClassType} 
+  deriving (Show)
+-- ClassType data type for specifying class type (Haskell, Drasil or GOOL)
+data ClassType = Haskell | Drasil | GOOL
+
+-- Eq instance of ClassType to enable comparisons
+instance Eq ClassType where
```

Comment:
This instance of `Eq` can probably be replaced by adding `deriving Eq` to `ClassType` definition above.